### PR TITLE
Publish API docs via CircleCI

### PR DESCRIPTION
**Pull request by erik-whiting**
- [ ] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [ ] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #...

### CODE REVIEW

File: .github/deploy_docs.sh [hunk 1]
```diff
@@ -5,65 +5,70 @@ if [[ $(git rev-parse --abbrev-ref HEAD) != "master" ]]; then
     exit 0
 fi
 
-# Assumes being called from the Biopython repository's root folder,
```
Comment (peterjc) [inline .github/deploy_docs.sh]: I would retain comment lines 8 to 15 about this script being run after sphinx-build

File: .github/deploy_docs.sh [hunk 2]
```diff
@@ -5,65 +5,70 @@ if [[ $(git rev-parse --abbrev-ref HEAD) != "master" ]]; then
...
+#     cat biopython_doc_key
+# and copy the contents. Back in the CircleCI UI, paste the contents
+# you just copied into the "Private Key" text box. Click "Add SSH Key".
 #
-# Finally, we add the private key to TravisCI by going to
-# https://travis-ci.org/biopython/biopython/settings or any authorised
-# fork like https://travis-ci.org/peterjc/biopython/settings and
-# setting DOC_KEY to the following (secret) value:
+# We now have to set an environment variable called DOC_KEY
+# in the Biopython project seetings. To get this value in a format
```
Comment (peterjc) [inline .github/deploy_docs.sh]: Typo: settings

File: .gitignore [hunk 3]
```diff
@@ -122,3 +122,9 @@ Doc/api/*.rst
 
 #Ignore mypy cache directory
 .mypy_cache
+
+#Ignore virtual environment file
+.env
+
+#Ignore keys for publishing docs
+biopython_doc_key*
```
Comment (peterjc) [inline .gitignore]: Should probably add ``biopython_doc_key.pub`` too, but this hardly seem worth doing to me. This should only exist on one developer's machine - and most likely won't even be in this directory (I used to use ``~/.ssh/`` for this).

File: .github/deploy_docs.sh [hunk 4]
```diff
@@ -134,13 +136,16 @@ cp -R $SOURCE_DIR/* $DEST_DIR/
...
+
 if [[ -z $(git status --porcelain) ]]; then
     echo "Nothing has changed, nothing needs pushing."
 else
     echo "Making commit of new files"
     git config user.email "sphinx@example.org"
     git config user.name "Sphinx"
-    git commit -m "Automated update ${TRAVIS_COMMIT:-}"
+    COMMIT_HASH=$(git rev-parse HEAD)
+    git commit -m "Automated update ${COMMIT_HASH}"
```
Comment (peterjc) [inline .github/deploy_docs.sh:157]: Sadly that will give the most recent git commit on the documentation repository, e.g. the commit comment for https://github.com/erik-whiting/docs/commit/ba92d210963f117487d2088d3dcb619e174701d2 points to https://github.com/erik-whiting/docs/commit/b6d8aef0eaead6759ef239dcb057ebf4ddb117ba

You can probably move the ``COMMIT_HASH=$(git rev-parse HEAD)`` line up the script before leaving the clone of the source code repository?

**Review by peterjc — CHANGES_REQUESTED** [review]
The git commit message needs tweaking (see in-line comment).

A lot of the changes are on the comments / human procedure. Was there a vital change in the script itself as well, or was it most likely I missed a set on the CircleCI side when I tried this before?

File: .gitignore (hunk 3 above)
Comment (peterjc) [reply .gitignore]: i.e. Ignore both key files, or don't add either of them.

**Comment by peterjc** [comment]
I see you got it to work on https://github.com/erik-whiting/docs which is great.

Note you will need to agree to the licensing before we merge this.

**Comment by peterjc** [comment]
Sorry, I missed this was a PR into your own repository - the one I should be focussed on is https://github.com/biopython/biopython/pull/3787

**Comment by erik-whiting** [comment]
Merged in the main project

**Pull request closed without merging**